etcdserver/server.go

```diff
@@ -143,14 +144,28 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 	var s *raft.MemoryStorage
 	var id types.ID
 
-	walVersion, err := wal.DetectVersion(cfg.DataDir)
+	dataVer, err := version.DetectDataDir(cfg.DataDir)
 	if err != nil {
 		return nil, err
 	}
-	if walVersion == wal.WALUnknown {
-		return nil, fmt.Errorf("unknown wal version in data dir %s", cfg.DataDir)
+	// Run the migrations.
+	if err := upgradeDataDir(cfg.DataDir, cfg.Name, dataVer); err != nil {
+		return nil, err
 	}
-	haveWAL := walVersion != wal.WALNotExist
+	if dataVer == version.DataDirUnknown {
+		names, err := fileutil.ReadDir(cfg.DataDir)
+		if err != nil {
+			return nil, err
+		}
+		for _, name := range names {
+			if name == "member" || name == "proxy" {
+				continue
+			}
+			log.Printf("etcdserver: extra file %s found in data-dir %s", name, cfg.DataDir)
```

Comment:
Why do we need to log it? If we really must log it, we should let the user know we plan to ignore the file.

```
etcdserver: ignoring extra file extra.txt found in data-dir /var/lib/etcd
```